app/uk/gov/hmrc/helptosavefrontend/models/account/AccountNumber.scala

```diff
@@ -18,7 +18,7 @@ package uk.gov.hmrc.helptosavefrontend.models.account
 
 import play.api.libs.json.{Format, Json}
 
-case class AccountNumber(accountNumber: String)
+case class AccountNumber(accountNumber: Option[String])
```

Comment:
Now that AccountNumber holds an option of a String, is there a good reason for uk.gov.hmrc.helptosavefrontend.services.HelpToSaveServiceImpl.SubmissionSuccess to contain an Option[AccountNumber], rather than just an AccountNumber ?

Reply:
Doing that messes up the createAccount method in RegisterController, and I'm not sure how to re-work it

Reply:
Having made:

case class SubmissionSuccess(accountNumber: AccountNumber) extends SubmissionResult

and made helpToSaveService.createAccount return SubmissionSuccess(AccountNumber(None)) instead of SubmissionSuccess(None)

then in RegisterController.createAccount you could just _.accountNumber.fold in place of _.fold( at line 150

I think

Reply:
traverse is no longer available on line 128

Reply:
How about as in my https://github.com/hmrc/help-to-save-frontend/tree/dave/DLS-245/incorrect-payment-ref

Reply:
I dont think that will compile